digitalocean/database/resource_database_kafka_topic.go

```diff
@@ -0,0 +1,573 @@
+							Optional:     true,
+							ForceNew:     false,
+							ValidateFunc: validateUint64(),
+						},
+						"unclean_leader_election_enable": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							ForceNew: false,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceDigitalOceanDatabaseKafkaTopicCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+	clusterID := d.Get("cluster_id").(string)
+
+	partition_count := uint32(d.Get("partition_count").(int))
+	replication_factor := uint32(d.Get("replication_factor").(int))
+
+	opts := &godo.DatabaseCreateTopicRequest{
+		Name:              d.Get("name").(string),
+		PartitionCount:    &partition_count,
+		ReplicationFactor: &replication_factor,
+	}
+
+	if v, ok := d.GetOk("config"); ok {
+		opts.Config = getTopicConfig(v.([]interface{}))
+	}
+
+	log.Printf("[DEBUG] Database kafka topic create configuration: %#v", opts)
+	topic, _, err := client.Databases.CreateTopic(context.Background(), clusterID, opts)
+	if err != nil {
+		return diag.Errorf("Error creating database kafka topic: %s", err)
+	}
+
+	d.SetId(makeDatabaseTopicID(clusterID, topic.Name))
+	log.Printf("[INFO] Database kafka topic name: %s", topic.Name)
+
+	return resourceDigitalOceanDatabaseKafkaTopicRead(ctx, d, meta)
+}
+
+func resourceDigitalOceanDatabaseKafkaTopicUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+	clusterID := d.Get("cluster_id").(string)
+
+	topicName := d.Get("name").(string)
+	partition_count := uint32(d.Get("partition_count").(int))
+	replication_factor := uint32(d.Get("replication_factor").(int))
+
+	opts := &godo.DatabaseUpdateTopicRequest{
+		PartitionCount:    &partition_count,
+		ReplicationFactor: &replication_factor,
+	}
+
+	if v, ok := d.GetOk("config"); ok {
+		opts.Config = getTopicConfig(v.([]interface{}))
+	}
+
+	log.Printf("[DEBUG] Database kafka topic update configuration: %#v", opts)
+	_, err := client.Databases.UpdateTopic(context.Background(), clusterID, topicName, opts)
+	if err != nil {
+		return diag.Errorf("Error updating database kafka topic: %s", err)
+	}
+
+	return resourceDigitalOceanDatabaseKafkaTopicRead(ctx, d, meta)
+}
+
+func resourceDigitalOceanDatabaseKafkaTopicRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+	clusterID := d.Get("cluster_id").(string)
+	topicName := d.Get("name").(string)
+
+	topic, resp, err := client.Databases.GetTopic(ctx, clusterID, topicName)
+	if err != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.Errorf("Error retrieving kafka topic: %s", err)
+	}
+
+	d.Set("state", topic.State)
+	d.Set("replication_factor", topic.ReplicationFactor)
+	d.Set("partitions", topic.Partitions)
```

Comment:
Similar issue here:

```
panic: Invalid address to set: []string{"partitions"}

goroutine 729 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc000995e80, {0x113e468, 0xa}, {0xf736c0, 0xc0008ed938})
        /home/asb/projects/go/src/github.com/digitalocean/terraform-provider-digitalocean/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:230 +0x2b5
github.com/digitalocean/terraform-provider-digitalocean/digitalocean/database.resourceDigitalOceanDatabaseKafkaTopicRead({0x146b5b8, 0xc0008eb260}, 0x37?, {0xff81e0?, 0xc0004bb320?})
        /home/asb/projects/go/src/github.com/digitalocean/terraform-provider-digitalocean/digitalocean/database/resource_database_kafka_topic.go:371 +0x25e
```

Reply:
Fixed, I removed `partitions` as it is not relevant for configuration but rather shows state on each partition. It is probably not relevant for the TF client